File: 1516-the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n.swift
class Solution {
    func getHappyString(_ n: Int, _ k: Int) -> String {
    
    let character = ["a", "b", "c"]

    func gen(_ pref: String = "") -> [String] {
        pref.count == n ? [pref] : character
            .filter { pref.isEmpty || $0 != String(pref.last!) }
            .map { pref + $0 }
            .flatMap { gen($0) }
    }

    let strs = gen()

    return k > strs.count ? "" : strs[k - 1]
        
    }
}

File: 1516-the-k-th-lexicographical-string-of-all-happy-strings-of-length-n/the-k-th-lexicographical-string-of-all-happy-strings-of-length-n.kt
	class Solution {
		fun getHappyString(n: Int, k: Int): String {
			val res = StringBuilder()
			val chars = charArrayOf('a', 'b', 'c')
			var count = 0
			fun dfs(s: String) {
				if (s.length == n) {
					count++
					if (count == k) {
						res.append(s)
					}
					return
				}
				for (c in chars) {
					if (s.isEmpty() || s.last() != c) {
						dfs(s + c)
					}
				}
			}
			dfs("")
			return res.toString()
		}
	}